### PR TITLE
fix(gnovm):  make stacktrace correctly handle realm panic

### PR DESCRIPTION
**Pull request by ltzmaxwell**
While finalizing the realm, all statements in the machine have been popped out. A necessary check must be performed during stack trace handling.

<details><summary>Contributors' checklist...</summary>

- [ ] Added new tests, or not needed, or not feasible
- [ ] Provided an example (e.g. screenshot) to aid review or the PR is self-explanatory
- [ ] Updated the official documentation or not needed
- [ ] No breaking changes were made, or a `BREAKING CHANGE: xxx` message was included in the description
- [ ] Added references to related issues and PRs
- [ ] Provided any useful hints for running manual tests
</details>

### CODE REVIEW

**Comment by Gno2D2** [comment]
I'm a bot that assists the Gno Core team in maintaining this repository. My role is to ensure that contributors understand and follow our guidelines, helping to streamline the development process.

The following requirements must be fulfilled before a pull request can be merged.
Some requirement checks are automated and can be verified by the CI, while others need manual verification by a staff member.

These requirements are defined in this [configuration file](https://github.com/gnolang/gno/tree/master/contribs/github-bot/internal/config/config.go).

## Automated Checks

 🟢 Maintainers must be able to edit this pull request
 🔴 The pull request head branch must be up-to-date with its base


## Manual Checks

*No manual checks match this pull request.*

<details><summary><b>Debug</b></summary><blockquote>
<details><summary><b>Automated Checks</b></summary><blockquote>

<details><summary>Maintainers must be able to edit this pull request</summary><blockquote>

### If
```
🟢 Condition met
└── 🟢 On every pull request

```
### Then
```
🟢 Requirement satisfied
└── 🟢 Maintainer can modify this pull request

```
</blockquote></details>

<details><summary>The pull request head branch must be up-to-date with its base</summary><blockquote>

### If
```
🟢 Condition met
└── 🟢 On every pull request

```
### Then
```
🔴 Requirement not satisfied
└── 🔴 Head branch (ltzmaxwell:fix/maxwell/stacktrace) is up to date with base (master): behind by 24 / ahead by 1

```
</blockquote></details>

</blockquote></details>



</blockquote></details>

**Comment by leohhhn** [comment]
Hey @ltzmaxwell, in what case does this problem show up? Can you add some example stack before and after? Just trying to figure out if this will change any edge case for users.

**Comment by ltzmaxwell** [comment]
> Hey @ltzmaxwell, in what case does this problem show up? Can you add some example stack before and after? Just trying to figure out if this will change any edge case for users.

Hey @leohhhn 

this is what it looks before:
```
 -test.short -run 'TestFiles$/realm14.gno' -v -p 1 -timeout=30m
=== RUN   TestFiles
=== PAUSE TestFiles
=== CONT  TestFiles
=== RUN   TestFiles/realm14.gno
--- FAIL: TestFiles (0.07s)
    --- FAIL: TestFiles/realm14.gno (0.02s)
panic: deleted descendants should not have a reference count of less than zero [recovered]
	panic: deleted descendants should not have a reference count of less than zero [recovered]
	panic: runtime error: index out of range [-1] [recovered]
	panic: runtime error: index out of range [-1]

goroutine 24 [running]:
testing.tRunner.func1.2({0x204c320, 0xc0006986f0})
	/usr/local/go/src/testing/testing.go:1631 +0x24a
testing.tRunner.func1()
	/usr/local/go/src/testing/testing.go:1634 +0x377
```

this is what it looks now:
```
 -test.short -run 'TestFiles$/realm14.gno' -v -p 1 -timeout=30m
=== RUN   TestFiles
=== PAUSE TestFiles
=== CONT  TestFiles
=== RUN   TestFiles/realm14.gno
    files_test.go:92: unexpected panic: deleted descendants should not have a reference count of less than zero
        output:

        stack:
        goroutine 37 [running]:
        runtime/debug.Stack()
        	/usr/local/go/src/runtime/debug/stack.go:24 +0x5e
        github.com/gnolang/gno/gnovm/pkg/test.(*TestOptions).runTest.func1()
        	/Users/maxwell/workbench/blockchain/gno/gno_core/gno/gnovm/pkg/test/filetest.go:196 +0xbf
        panic({0x203a800?, 0x2122f60?})
        	/usr/local/go/src/runtime/panic.go:770 +0x132
        github.com/gnolang/gno/gnovm/pkg/gnolang.(*Machine).Run.func1()
        	/Users/maxwell/workbench/blockchain/gno/gno_core/gno/gnovm/pkg/gnolang/machine.go:1140 +0x9d
        panic({0x203a800?, 0x2122f60?})
        	/usr/local/go/src/runtime/panic.go:770 +0x132
        github.com/gnolang/gno/gnovm/pkg/gnolang.(*Realm).decRefDeletedDescendants(0xc00015cee0, {0x49dc2dc8, 0xc00051b220}, {0x2139d80, 0xc000572fa0})
```

**Review by petar-dambovaliev — APPROVED** [review]

**Review by thehowl — APPROVED** [review]

**Review by omarsy — APPROVED** [review]

**Pull request merged**
